core/trino-main/src/main/java/io/trino/execution/scheduler/FallbackToFullNodePartitionMemoryEstimator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler;
+
+import io.airlift.units.DataSize;
+import io.trino.Session;
+import io.trino.spi.ErrorCode;
+
+import static io.trino.spi.StandardErrorCode.EXCEEDED_LOCAL_MEMORY_LIMIT;
+
+public class FallbackToFullNodePartitionMemoryEstimator
```

Comment:
For the future. This seems like a dedicated estimator for specific node allocator. Does it make sense to separate these? I imagine retires and allocation strategy is correlated how we estimate memory (e.g. how we increase estimates on subsequent failures). This is somehow related to https://github.com/trinodb/trino/pull/10432#discussion_r804628055 and how node allocation could actually keep track of `Task` handles and do estimates itself.